platform/darwin/docs/guides/Data-Driven Styling.md.ejs

```diff
@@ -0,0 +1,122 @@
+
+### Stops
+
+Stops are key-value pairs that that determine a style value. With a `MGLCameraSourceFunction` stop, you can use a dictionary with a zoom level for a key and a `MGLStyleValue` for the value. For example, you can use a stops dictionary with zoom levels 0, 10, and 20 as keys, and yellow, orange, and red as the values. A `MGLSourceStyleFunction` uses the relevant attribute value as the key.
+
+### Interpolation Mode
+
+The effect a key has on the style value is determined by the interpolation mode. There are four interpolation modes that can be used with a source style function: exponential, interval, categorical, and identity. You can also use exponential and interval interpolation modes with a camera style function.
+
+#### Exponential (or Linear)
+
+`MGLInterpolationModelExponential` creates a linear or exponential effect based on the values. The key value is the starting point for interpolation, and the style value is based on where an attribute value falls between two keys. By default, `MGLInterpolationModeExponential` uses an interpolation base of 1. This causes a linear relationship between the style value and stop key.
+
+The stops dictionary below, for example, shows colors that continuously shift from yellow to orange to red to blue to white based on the attribute value.
+
+``` swift
+    let url = URL(string: "https://earthquake.usgs.gov/earthquakes/feed/v1.0/summary/all_week.geojson")
+    let symbolSource = MGLSource(identifier: "source")
+    let symbolLayer = MGLSymbolStyleLayer(identifier: "place-city-sm", source: symbolSource)
+
+    let source = MGLShapeSource(identifier: "earthquakes", url: url, options: nil)
+    style.addSource(source)
+
+    let stops = [0 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.yellow),
+                 2.5 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.orange),
+                 5: MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.red),
+                 7.5 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.blue),
+                 10 : MGLStyleValue(rawValue: <%- cocoaPrefix %>Color.white)]
+
+    let layer = MGLCircleStyleLayer(identifier: "circles", source: source)
+    layer.circleColor = MGLStyleValue(interpolationMode: .exponential,
+                                      sourceStops: stops,
+                                      attributeName: "mag",
+                                      options: [.defaultValue: MGLStyleValue<<%- cocoaPrefix %>Color>(rawValue: .green)])
+    layer.circleRadius = MGLStyleValue(rawValue: 10)
+    style.insertLayer(layer, below: symbolLayer)
+```
+
+![exponential mode](img/data-driven-styling/exponential.png)
+
+#### Interval
+
+`MGLInterpolationModeInterval` creates a range using the keys stops dictionary. The range is from the given key to just less than the next key. The attribute values that fall into that range are then styled using the style value assigned to that key.
```

Comment:
>keys stops dictionary

Is this a dictionary of stops for multiple keys? Pluralization is a little awkward, but if the meaning is correct then it’s probably OK.